packages/es-client/esScrollSearch.js

```diff
@@ -35,28 +35,37 @@ const {
  *
  */
 class ESScrollSearch extends Search {
-  async query() {
+  async query(searchContext) {
     if (!this.client) {
       this.client = await super.constructor.es();
     }
     let response;
-    if (!this.scrollId) {
+    if (!this.scrollId && !searchContext) {
       const searchParams = this._buildSearch();
-      searchParams.size = Number(process.env.ES_SCROLL_SIZE || defaultESScrollSize);
+      // this.params.limit exists for API invocation vs. not for reconciliation usage
+      searchParams.size = Number(this.params.limit
+        ? this.size : (process.env.ES_SCROLL_SIZE || defaultESScrollSize));
       searchParams.scroll = process.env.ES_SCROLL || defaultESScrollDuration;
       response = await this.client.search(searchParams);
       this.scrollId = response.body._scroll_id;
     } else {
       response = await this.client.scroll({
-        scrollId: this.scrollId,
+        scrollId: searchContext || this.scrollId,
```

Comment:
question about `searchContext` vs `scrollId`:
`searchContext` is set to `sort` object in search.js and esSearchAfter.js, and is set to `scrollId` here. If we want to return `searchContext`, should we return JSON.stringify({ scrollId }) instead?
Is there a use case to use query(searchContext)?

Reply:
No, this is left over from the initial scroll-based implementation. It does still need an update to stringify and parse, but I can also take it out since we are not using it. Otherwise however, we could also implement a scroll-based endpoint with these changes so I am mixed on removing it entirely. Scrolls are, according to ES, not suitable for real-time user requests because maintaining the scroll context is expensive. However, unlike search-after, the scroll context is static as opposed to dynamically updated with the index as search-after is so there are some benefits to keeping the option open. Thoughts?

Reply:
I would suggest to remove the changes from this file for now if we think `search-after` is the preferred way for the end user.

Reply:
All scroll changes removed in https://github.com/nasa/cumulus/pull/2993/commits/92ec3147d06575b9d3f911f98694f115b03cd676